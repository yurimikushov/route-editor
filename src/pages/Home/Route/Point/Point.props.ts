import { Address } from 'services/routeEditor'

type PointProps = {
  className?: string
  isDraggable?: boolean
  point: Address
  onDelete: () => void
}

export default PointProps
