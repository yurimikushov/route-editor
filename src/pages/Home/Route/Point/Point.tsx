import { FC } from 'react'
import cn from 'classnames'
import ClearButton from 'components/ClearButton'
import PointProps from './Point.props'

const Point: FC<PointProps> = ({ className, point, onDelete }) => {
  return (
    <div
      className={cn(
        className,
        'my-3 px-3 py-2',
        'flex justify-between items-center',
        'bg-gray-100 rounded'
      )}
    >
      <div className='flex flex-col'>
        <span className='text-gray-600'>{point.name}</span>
        <span className='text-sm text-gray-400'>{point.description}</span>
      </div>
      <ClearButton className='w-5 h-5 min-w-5 min-h-5' onClick={onDelete} />
    </div>
  )
}

export default Point
