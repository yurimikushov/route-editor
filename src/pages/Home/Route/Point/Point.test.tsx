import renderer from 'react-test-renderer'
import noop from 'lodash/noop'
import Point from './Point'

// eslint-disable-next-line react/display-name
jest.mock('components/Dragger', () => (props: { className?: string }) => (
  <div {...props}>Dragger</div>
))

// eslint-disable-next-line react/display-name
jest.mock('components/ClearButton', () => (props: { className?: string }) => (
  <button {...props}>Clear</button>
))

it('should render non-draggable point by default', () => {
  const tree = renderer
    .create(
      <Point
        point={{
          id: '2',
          name: 'Vodka',
          description: 'Some description',
          point: {
            lat: 0,
            lon: 0,
          },
        }}
        onDelete={noop}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render draggable point when isDraggable prop is true', () => {
  const tree = renderer
    .create(
      <Point
        isDraggable
        point={{
          id: '1',
          name: 'Balalayka',
          description: 'Some description',
          point: {
            lat: 0,
            lon: 0,
          },
        }}
        onDelete={noop}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render non-draggable point when isDraggable prop is false', () => {
  const tree = renderer
    .create(
      <Point
        isDraggable={false}
        point={{
          id: '1',
          name: 'Balalayka',
          description: 'Some description',
          point: {
            lat: 0,
            lon: 0,
          },
        }}
        onDelete={noop}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
