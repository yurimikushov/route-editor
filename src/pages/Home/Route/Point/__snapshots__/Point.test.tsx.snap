// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render point 1`] = `
<div
  className="my-3 px-3 py-2 flex justify-between items-center bg-gray-100 rounded"
>
  <div
    className="flex flex-col"
  >
    <span
      className="text-gray-600"
    >
      Balalayka
    </span>
    <span
      className="text-sm text-gray-400"
    >
      Some description
    </span>
  </div>
  <button
    className="w-5 h-5 min-w-5 min-h-5"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;

exports[`should render point 2`] = `
<div
  className="my-3 px-3 py-2 flex justify-between items-center bg-gray-100 rounded"
>
  <div
    className="flex flex-col"
  >
    <span
      className="text-gray-600"
    >
      Vodka
    </span>
    <span
      className="text-sm text-gray-400"
    >
      Some description
    </span>
  </div>
  <button
    className="w-5 h-5 min-w-5 min-h-5"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;
